### PR TITLE
Contain custom image size and add a link to show them full size in another tab

### LINKED ISSUE #5086: Wide image destroys the case page layout
**Describe the bug**

When the image is wide enough, it pushes itself to a new row, leaving the sidebar behind.

**To Reproduce**

Upload a wide enough image with a case and toggle it open on the case page.

**Expected behavior**

Don't break the layout. Probably a scroll-box would be my recommended approach.

**Screenshots**

large_image_destroys_layout

**Additional context**

I would be happy to provide a PR for this, if you agree with this issue (as it is easier for me to test, having the image already available).

### PR DESCRIPTION
**Pull request by northwestwitch**
This PR adds a functionality or fixes a bug.
- Closes #5086

## Image gets resized with ratio in scout:

image

## New tab with the full size image:

image

<details>
<summary>Testing on cg-vm1 server (Clinical Genomics Stockholm)</summary>

**Prepare for testing**
1. Make sure the PR is pushed and available on [Docker Hub](https://hub.docker.com/repository/docker/clinicalgenomics/scout-server-stage)
1. Fist book your testing time using the Pax software available at [https://pax.scilifelab.se/](https://pax.scilifelab.se). The resource you are going to call dibs on is `scout-stage` and the server is `cg-vm1`.
1. `ssh <USER.NAME>@cg-vm1.scilifelab.se`
1. `sudo -iu hiseq.clinical`
1. `ssh localhost`
1. (optional) Find out which scout branch is currently deployed on cg-vm1: `podman ps`
1. Stop the service with current deployed branch: `systemctl --user stop scout.target`
1. Start the scout service with the branch to test: `systemctl --user start scout@<this_branch>`
1. Make sure the branch is deployed: `systemctl --user status scout.target`
1. After testing is done, repeat procedure at [https://pax.scilifelab.se/](https://pax.scilifelab.se), which will release the allocated resource (`scout-stage`) to be used for testing by other users.
</details>

<details>
<summary>Testing on hasta server (Clinical Genomics Stockholm)</summary>

**Prepare for testing**
1. `ssh <USER.NAME>@hasta.scilifelab.se`
1. Book your testing time using the Pax software. `us; paxa -u <user> -s hasta -r scout-stage`. You can also use the WSGI Pax app available at [https://pax.scilifelab.se/](https://pax.scilifelab.se).
1. (optional) Find out which scout branch is currently deployed on cg-vm1: `conda activate S_scout; pip freeze | grep scout-browser`
1. Deploy the branch to test: `bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_scout -t scout -b <this_branch>`
1. Make sure the branch is deployed: `us; scout --version`
1. After testing is done, repeat the `paxa` procedure, which will release the allocated resource (`scout-stage`) to be used for testing by other users.
</details>

**How to test**:
1. Add a very big custom image to case page or a variant page
2. Make sure the page doesn't break
3. Click on the image and check that the image gets displayed much bigger on a new tab

**Expected outcome**:
The functionality should be working
Take a screenshot and attach or copy/paste the output.

**Review:**
- [ ] code approved by
- [ ] tests executed by

### CODE REVIEW

**Comment by northwestwitch** [comment]
I have a PR to fix that image size issue @Jakob37, you could perhaps review?

**Comment by Jakob37** [comment]
> I have a PR to fix that image size issue @Jakob37, you could perhaps review?

Yes, I'll check!

**Comment by Jakob37** [comment]
Hmm. I think this would work nicely on a large display, but less so on my laptop (where I am testing it out right now). I suspect it is a very normal size for a laptop (1920 x 1080, 150% scaled, both at recommended-by-windows settings).

Confirming that the image indeed is resized to 1200px:

wide_image_4

Still shifted to a new row:

wide_image_3

Checking with the inspector, it looks like I have 992px space (inside the padding) in the area right of the left bar.

A basic solution would maybe be to reduce the width even further to 950px or so.

A more correct solution would probably be to see if you could make the image fit whatever size is available beyond the left sidebar. This would require more CSS-digging though. Not sure whether that currently is worth it or not for you. Up to you :)

File: scout/server/static/custom_images.js [hunk 1]
```diff
@@ -9,15 +17,24 @@
 		 .then(blob => {
 				 const img = document.createElement("img");
 				 img.src = URL.createObjectURL(blob);
+				 if (width > imgMaxWidth || height> imgMaxHeight) {
```
Comment (Jakob37) [inline scout/server/static/custom_images.js:20]:
```suggestion
				 if (width > imgMaxWidth || height > imgMaxHeight) {
```

File: scout/server/static/custom_images.js [hunk 2]
```diff
@@ -9,15 +17,24 @@
...
+				 		var resized_pixels = calculateAspectRatioFit(width, height, imgMaxWidth, imgMaxHeight);
```
Comment (Jakob37) [inline scout/server/static/custom_images.js:21]:
```suggestion
				 		const resized_pixels = calculateAspectRatioFit(width, height, imgMaxWidth, imgMaxHeight);
```

I would stick with the block scoped `let` and `const` of you don't have specific reasons not to. Here the value is not mutated, so `const` would be the correct one.

**Review by Jakob37 — COMMENTED** [review]
It does not solve it yet on a small laptop screen.

The ideal would be to solve this without hard-coded widths and heights, using CSS. But that is more intrusive and maybe beyond the scope of the current work.

**Comment by Jakob37** [comment]
Opening in new window on click works nicely, that will be convenient!

**Comment by northwestwitch** [comment]
> It does not solve it yet on a small laptop screen (see comment above).
> 
> The ideal would be to solve this without hard-coded widths and heights, using CSS. But that is more intrusive and maybe beyond the scope of the current work. Up to you!

I'll see if these max width and height can be set automatically given the browser's window size. I agree that that way is better 👍🏻

**Comment by northwestwitch** [comment]
I've pushed a possible fix, when you have some time, check how it works for you!

**Review by Jakob37 — APPROVED** [review]
Looks good to me!

**Comment by northwestwitch** [comment]
Thanks for the review @Jakob37. I'm slightly modified the code to replace one of the 2 jpeg images (small images, same size) of the demo case with a bigger one. Merging when tests are passing!

**Pull request merged**